### ISSUE TITLE
JavaScript permission option missing from Site settings when accessed through the address bar

### ISSUE THREAD

**Issue by Master0ne** [bromite/bromite#990]
### Bromite version
Version: `89.0.4389.78`
Arch: `arm64`
Android version: `11.0`
Device model: `POCO X3 NFC (surya)`

### Is this bug about the SystemWebView?
No

### Is the bug reproducible with latest version?
Yes

### Can the bug be reproduced with corresponding Chromium version?
Unknown

### Allow ads for the site; is the bug caused by the adblocker?
No

### Is the bug a crash?
No

### Describe the bug
In previous versions it was always possible to set the JavaScript option in the Site settings when accessed through the address bar and that option is gone with the latest release (Permissions only showing `Cookies`, `Ads` and `Sound` now), so with JavaScript blocked globally one has to access Settings >> Site settings >> JavaScript >> ADD SITE EXCEPTION with entering the site URL to allow JavaScript individually instead of previously just clicking at the option when accessed through the address bar.

### Steps to reproduce the bug
Steps to reproduce the bug:
1. Disable JavaScript in Settings >> Site Settings >> JavaScript.
2. Visit any website.
3. Click on the lock icon at the left of the address bar and then `Site settings`.
4. The JavaScript option is not shown anymore under `Permissions`.

### Expected behavior
The Site settings should show JavaScript with the option to `Allow` or `Block` under Permissions when accessed from the address bar.

### Screenshots
![screenshot](https://user-images.githubusercontent.com/8912853/110422047-ec257d00-809e-11eb-9bf8-52313d388ab9.jpg)

**Comment by uazo** [comment]
the bug is here

```
-        // Not possible to embargo JAVASCRIPT.
-        setupContentSettingsPreference(preference, currentValue, false /* isEmbargoed */);
+        setUpListPreference(preference, currentValue, false);
```

that must be revert (actual chromium code is ok)
if you want I have the patch ready.

**Comment by csagan5** [comment]
I wish I had seen you comment earlier; by the way @uazo  what do you think of these checks?
```
@@ -364,6 +364,12 @@ public class SingleWebsiteSettings extends SiteSettingsPreferenceFragment
                 merged.setContentSettingException(ContentSettingsType.ADS,
                         other.getContentSettingException(ContentSettingsType.ADS));
             }
+            if (merged.getContentSettingException(ContentSettingsType.JAVASCRIPT) == null
+                    && other.getContentSettingException(ContentSettingsType.JAVASCRIPT) != null
+                    && other.compareByAddressTo(merged) == 0) {
+                merged.setContentSettingException(ContentSettingsType.JAVASCRIPT,
+                        other.getContentSettingException(ContentSettingsType.JAVASCRIPT));
+            }
             for (PermissionInfo info : other.getPermissionInfos()) {
                 if (merged.getPermissionInfo(info.getContentSettingsType()) == null
                         && permissionInfoIsForTopLevelOrigin(info, origin)) {
@@ -392,6 +398,9 @@ public class SingleWebsiteSettings extends SiteSettingsPreferenceFragment
                     if (type == ContentSettingsType.ADS) {
                         continue;
                     }
+                    if (type == ContentSettingsType.JAVASCRIPT) {
+                        continue;
+                    }
                     if (merged.getContentSettingException(type) == null) {
                         merged.setContentSettingException(type, exception);
                     }
```
I was experimenting with these.
Should we have them for Javascript (and cookies too)? I am thinking that it is odd that the check happens only for `ADS`

**Comment by uazo** [comment]
I do not think it is very important, what I read only serves to minimize the space used by the exceptions of the content settings, in case there are overlapped rules.
I wouldn't worry about it, considering all the todo there are.

by the way the relative commit says:

**Move website settings fetching from WebsitePreferences to new WebsitePermissionFetcher.**
https://source.chromium.org/chromium/chromium/src/+/ac1a3823f82e90136e7800d6f77a08534f9afbd1
it was probably only useful in that case

It is also used in PageInfoV2, which is very nice, but only as a display, I'd like to check how it works and possibly propose it by default

**Comment by csagan5** [comment]
Fixed in `89.0.4389.92`.
